superset/charts/data/api.py

```diff
@@ -221,6 +221,7 @@ def data(self) -> Response:
         if json_body is None:
             return self.response_400(message=_("Request is not JSON"))
 
+        form_data = json_body.pop("form_data", None)
```

Comment:
Do we need to pop here? Can't we just do
```suggestion
        form_data = json_body.get("form_data")
```

Reply:
We don't need form_data in json_body. We pass it as a separate argument

Reply:
> > A few small comments, but this works and looks great! It would be nice if we could add a few tests:
> > 
> > 1. Ensure that a query context with `form_data` validates successfully
> > 2. Add tests for `ExportToCSVDropdown` that ensures it renders correctly with standard/pivoted options + the children.
> 
> Added tests for `ExportToCSVDropdown`. I didn't add tests for query context validation, as we pop form_data from json_body before it's used to create context, so there's no change there.

Edit: After some discussion, I implemented Ville's suggestions. We already had schema validations tests in place, so I just added `form_data` to query context generator.